adb/adb_commands.py

```diff
@@ -148,15 +227,16 @@ def Push(self, source_file, device_filename, mtime='0', timeout_ms=None):
         for f in os.listdir(source_file):
           self.Push(os.path.join(source_file, f), device_filename + '/' + f)
         return
-      source_file = open(source_file)
+      source_file = open(source_file, 'rb')
 
-    connection = self.protocol_handler.Open(
+    with source_file:
```

Comment:
This is a bit cleaner than the current should_close variable solution in master. Allows an external caller to pass any file-like object and ensure its properly closed.